ch5-catdir/controller.cpp

```diff
@@ -26,13 +27,31 @@ Controller::~Controller()
 
 int Controller::run()
 {
-    if (m_args.empty())
+    if (m_args.m_wasError)
+    {
+        m_err << "Try '" << m_args.m_progName << " -h' for more information.\n";
+        return ExitStatus::CMDLINEERR;
+    }
+
+    if (m_args.m_showHelp)
+    {
+        m_err << "Usage: " << m_args.m_progName << " [-hr] [DIRECTORY]...\n"
+                 "Print the content of the DIRECTORY(ies).\n\n"
+                 "With no DIRECTORY, use the current working directory.\n"
+                 "Exit status is 0 for success, 1 for failure and 2 for"
+                 " command-line error.\n\n"
+                 "  -h\tShow this help and exit\n"
+                 "  -r\tAlso print the content of subdirectories\n";
```

Comment:
More commonly phrased as: `recurse into directories` (from `zip` and `tar` help)

Reply:
See my comment on the next conversation.